test/test_invalid_headers.py

```diff
@@ -256,8 +284,12 @@ def test_matching_authority_host_headers(self,
             (b':method', b'GET'),
             (b'host', b'example.com'),
         ]
-        assert headers == h2.utilities.validate_headers(
-            headers, hdr_validation_flags)
+        if not hdr_validation_flags.is_trailer:
```

Comment:
I think you just want to add a new set of header validation combos that doesn't include the trailer set, and use that here.